apps/courses/tests/test_factories.py

```diff
@@ -0,0 +1,79 @@
+"""
+Validate course specific factories logic
+"""
+
+import re
+
+from django.conf import settings
+from django.test import TestCase
+from django.utils import translation
+
+from apps.organizations.factories import OrganizationFactory
+
+from ..models import CourseOrganizationRelation
+from ..factories import CourseFactory, CourseSubjectFactory
+
+
+# This is the edX course_key pattern regexp
+COURSE_KEY_PATTERN = r"(?P<course_key_string>[^/+]+(/|\+)[^/+]+(/|\+)[^/?]+)"
```

Comment:
this regexp comes from edx code, and should also work for old (non splitmongo) course keys: ENSCachan/20004S02/session02
Also "session-" is  d2rp use of the third part of the key, it could be something else.